plugins/by-name/alpha/default.nix

```
@@ -135,77 +123,122 @@ in
-        '';
-      };
-
-      plugins.web-devicons =
-        lib.mkIf
-          (
-            opt.iconsEnabled.isDefined
-            && cfg.iconsEnabled
-            && !(
-              config.plugins.mini.enable
-              && config.plugins.mini.modules ? icons
-              && config.plugins.mini.mockDevIcons
-            )
-          )
+  settingsExample = {
+    layout = [
+      {
+        type = "padding";
+        val = 2;
+      }
+      {
+        type = "text";
+        val = [
+          "███╗   ██╗██╗██╗  ██╗██╗   ██╗██╗███╗   ███╗"
+          "████╗  ██║██║╚██╗██╔╝██║   ██║██║████╗ ████║"
+          "██╔██╗ ██║██║ ╚███╔╝ ██║   ██║██║██╔████╔██║"
+          "██║╚██╗██║██║ ██╔██╗ ╚██╗ ██╔╝██║██║╚██╔╝██║"
+          "██║ ╚████║██║██╔╝ ██╗ ╚████╔╝ ██║██║ ╚═╝ ██║"
+          "╚═╝  ╚═══╝╚═╝╚═╝  ╚═╝  ╚═══╝  ╚═╝╚═╝     ╚═╝"
+        ];
+        opts = {
+          position = "center";
+          hl = "Type";
+        };
+      }
+      {
+        type = "padding";
+        val = 2;
+      }
+      {
+        type = "group";
+        val = [
           {
-            enable = lib.mkOverride 1490 true;
-          };
+            type = "button";
+            val = "  New file";
+            on_press = nestedLiteralLua "function() vim.cmd[[ene]] end";
+            opts.shortcut = "n";
+          }
+          {
+            type = "button";
+            val = " Quit Neovim";
+            on_press = nestedLiteralLua "function() vim.cmd[[qa]] end";
+            opts.shortcut = "q";
+          }
+        ];
+      }
+      {
+        type = "padding";
+        val = 2;
+      }
+      {
+        type = "text";
+        val = "Inspiring quote here.";
+        opts = {
+          position = "center";
+          hl = "Keyword";
+        };
+      }
+    ];
+  };
+
+  extraOptions = {
+    theme = defaultNullOpts.mkStr' {
+      description = ''
+        The theme to use for the dashboard.
+      '';
+      example = "dashboard";
+      apply =
+        value: if builtins.isString value then mkRaw ''require("alpha.themes.${value}").config'' else value;
+    };
+  };
 
-      extraPlugins = [ cfg.package ];
+  optionsRenamedToSettings = [
+    "opts"
+    "layout"
+  ];
 
-      assertions = lib.nixvim.mkAssertions "plugins.alpha" [
+  extraConfig =
+    opts: cfg:
+    let
+      layoutDefined = cfg.settings.layout or null != null;
+      themeDefined = cfg.theme != null;
+    in
+    {
+      assertions = mkAssertions "plugins.alpha" [
         {
-          assertion = themeDefined || layoutDefined;
+          assertion = layoutDefined || themeDefined;
           message = ''
             You have to either set a `theme` or define some sections in `layout`.
           '';
         }
         {
-          assertion = !(themeDefined && layoutDefined);
+          assertion =
+            !(
+              themeDefined
+              && (opts.settings.highestPrio == 1500 && builtins.length opts.settings.definitions == 1)
+            );
```

I'm probably mis-understanding this logic. But should the assertion be `themeDefined` _implies_ `settings` not defined?

```suggestion
            themeDefined ->
              (opts.settings.highestPrio == 1500 && builtins.length opts.settings.definitions == 1);
```